usaspending_api/etl/elasticsearch_loader_helpers/delete_data.py

```diff
@@ -19,89 +19,46 @@
-    index: str,
-    max_query_size: int,
-    use_aliases: bool = False,
-    task_id: Optional[Tuple[int, str]] = None,
-) -> None:
-    """
-    id_list = [
-        {key: 'key1', col: 'transaction_id'},
-        {key: 'key2', col: 'generated_unique_transaction_id'},
-        ...
-    ]
-    - or -
-    id_list = [
-        {key: 'key1', col: 'award_id'},
-        {key: 'key2', col: 'generated_unique_award_id'},
-        ...
-    ]
-    """
-    start = perf_counter()
-    msg = f"Deleting up to {len(id_list):,} document{'s' if len(id_list) != 1 else ''}"
-    logger.info(format_log(msg, name=task_id, action="Delete"))
-
-    if use_aliases:
-        index = f"{index}-*"
-    start_ = client.count(index=index)["count"]
-    logger.info(format_log(f"Starting amount of indices ----- {start_:,}", name=task_id, action="Delete"))
-    col_to_items_dict = defaultdict(list)
-    for line in id_list:
-        col_to_items_dict[line["col"]].append(line["key"])
-
-    for column, values in col_to_items_dict.items():
-        logger.info(format_log(f"Deleting {len(values):,} of '{column}'", name=task_id, action="Delete"))
-        values_generator = chunks(values, 1000)
-        for v in values_generator:
-            # IMPORTANT: This delete routine looks at just 1 index at a time. If there are duplicate records across
-            # multiple indexes, those duplicates will not be caught by this routine. It is left as is because at the
-            # time of this comment, we are migrating to using a single index.
-            body = filter_query(column, v)
-            response = client.search(index=index, body=json.dumps(body), size=max_query_size)
-            delete_body = _delete_query(response)
-            try:
-                client.delete_by_query(index=index, body=json.dumps(delete_body), refresh=True, size=max_query_size)
-            except Exception:
-                logger.exception(format_log("", name=task_id, action="Delete"))
-                raise SystemExit(1)
-
-    end_ = client.count(index=index)["count"]
-    record_count = start_ - end_
-    duration = perf_counter() - start
-    msg = f"Delete operation took {duration:.2f}s. Removed {record_count:,} document{'s' if record_count != 1 else ''}"
-    logger.info(format_log(msg, name=task_id, action="Delete"))
-    return
-
-
-def delete_docs_by_unique_key(client: Elasticsearch, key: str, value_list: list, task_id: str, index) -> int:
+    key: str,
+    value_list: list,
+    task_id: str,
+    index,
+    refresh_after: bool = True,
+    delete_chunk_size: int = 10000,
+) -> int:
     """
     Bulk delete a batch of documents whose field identified by ``key`` matches any value provided in the
     ``values_list``.
 
+    NOTE: This delete routine looks at just the index name given. If there are duplicate records across
+    multiple indexes, an alias or wildcard should be provided for ``index`` param that covers multiple
+    indices, or this will need to be run once per index.
```

Comment:
Do deletes work for an alias with multiple indices?

Reply:
A couple searches would indicate that it can be done. Also, this was previous comment that already existed for the prior delete functionality. 

- https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-delete-by-query.html mentions use of an alias as the target
- old post when delete_by_query was a plugin shows support: https://discuss.elastic.co/t/delete-by-query-on-alias-or-multiple-indexes/43102/2
- alias docs: 
    - > Aliases with filters provide an easy way to create different "views" of the same index. The filter can be defined using Query DSL and is applied to all Search, Count, Delete By Query and More Like This operations with this alias.

Reply:
cool, thanks. I knew indexing documents will not work with multi-index aliases and now I know deletes are fine